Time system/Assets/Code/Scripts/Environment/DayNightCycle.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using Code.Scripts.Enums;
+using Code.Scripts.Environment.TimeSystem;
+using Code.Scripts.ScriptableObjects;
+using UnityEngine;
+
+namespace Code.Scripts.Environment
+{
+    [ExecuteAlways]
+    public class DayNightCycle : MonoBehaviour
+    {
+        private const float SunInitialIntensity = 1;
+
+        [SerializeField]
+        private TimeSettings timeSettings;
+
+        [SerializeField]
+        private TimeManager timeManager;
+
+        [SerializeField]
+        private Light sunLight;
+
+        [SerializeField]
+        private DayNightSettings dayNightSettings;
+
+        [SerializeField]
+        [Range(0, 24)]
+        private float timeOfDay;
+
+        // Internal use
+        private float internalTimeOfDay;
+
+        private float intensityMultiplier = .8f;

```

Comment:
Zou deze of de andere niet publiek moeten zijn? Zo kan het gameplay team hier een aanpassing aan doen indien nodig.

Reply:
Hierboven zit nog een SerializeField timeOfDay voor het gameplay team, alleen is deze waarde bedoelt voor **edit** mode. Het was eerder de bedoeling dat de start tijd gezet zou worden in de TimeSettings scriptable object.

Ik heb dit wat duidelijker gemaakt door in de inspector een header en tooltip toe te voegen.